modules/renter/hostdb/upload.go

```diff
@@ -201,17 +202,36 @@ outer:
 		}
 	}
 
-	// form new contracts from randomly-picked nodes
+	// Also exclude hosts on the pool's blacklist, and hosts we're already
```

Comment:
it's awkward to start this comment with also, I'm not sure what the previous thought was.

Probably sufficient to `Exclude...`